repository-data/webfiles/src/main/resources/site/freemarker/hee/pages/homepage-main.ftl

```diff
@@ -11,18 +11,21 @@
     <#assign heroType>
         class="nhsuk-hero-content"
     </#assign>
+    <#assign heroImgClass> nhsuk-hero--image nhsuk-hero--image-description</#assign>
```

Comment:
Could we default `heroImgClass` and `backgroundImage` variables to empty string to avoid the following errors:
```
[INFO] [talledLocalContainer] 09.12.2021 13:17:14 DEBUG http-nio-8080-exec-3 [BreadcrumbComponent.doBeforeRender:44] Breadcrumb Links will not be built as the current page is Home
[WARNING] [talledLocalContainer] %d{HH:mm:ss} WARN  [HstFreemarkerServlet] The following has evaluated to null or missing:
[INFO] [talledLocalContainer] ==> heroImgClass  [in template "webfile:/freemarker/hee/pages/homepage-main.ftl" at line 20, column 29]
[INFO] [talledLocalContainer]
[INFO] [talledLocalContainer] ----
[INFO] [talledLocalContainer] Tip: If the failing expression is known to legally refer to something that's sometimes null or missing, either specify a default value like myOptionalVar!myDefault, or use <#if myOptionalVar??>when-present<#else>when-missing</#if>. (These only cover the last step of the expression; to cover the whole expression, use parenthesis: (myOptionalVar.foo)!myDefault, (myOptionalVar.foo)??
[INFO] [talledLocalContainer] ----
[INFO] [talledLocalContainer]
[INFO] [talledLocalContainer] ----
[INFO] [talledLocalContainer] FTL stack trace ("~" means nesting-related):
[INFO] [talledLocalContainer] 	- Failed at: ${heroImgClass}  [in template "webfile:/freemarker/hee/pages/homepage-main.ftl" at line 20, column 27]
[INFO] [talledLocalContainer] ----. To see the stack trace, set 'org.hippoecm.hst.servlet.HstFreemarkerServlet' log-level to debug in log4j configuration or runtime via the logging servlet
[WARNING] [talledLocalContainer] %d{HH:mm:ss} WARN  [HstFreemarkerServlet] The following has evaluated to null or missing:
[INFO] [talledLocalContainer] ==> backgroundImage  [in template "webfile:/freemarker/hee/pages/homepage-main.ftl" at line 20, column 45]
[INFO] [talledLocalContainer]
[INFO] [talledLocalContainer] ----
[INFO] [talledLocalContainer] Tip: If the failing expression is known to legally refer to something that's sometimes null or missing, either specify a default value like myOptionalVar!myDefault, or use <#if myOptionalVar??>when-present<#else>when-missing</#if>. (These only cover the last step of the expression; to cover the whole expression, use parenthesis: (myOptionalVar.foo)!myDefault, (myOptionalVar.foo)??
[INFO] [talledLocalContainer] ----
[INFO] [talledLocalContainer]
[INFO] [talledLocalContainer] ----
[INFO] [talledLocalContainer] FTL stack trace ("~" means nesting-related):
[INFO] [talledLocalContainer] 	- Failed at: ${backgroundImage}  [in template "webfile:/freemarker/hee/pages/homepage-main.ftl" at line 20, column 43]
```
Also, could we use `<#assign heroImgClass='nhsuk-hero--image nhsuk-hero--image-description'>` (rather than `<#assign heroImgClass> nhsuk-hero--image nhsuk-hero--image-description</#assign>`) for best practices described in https://freemarker.apache.org/docs/ref_directive_assign.html [at the end of the page] ?

Reply:
Thanks for the pointer @Selva. Should be done.